features/upgrade/sdn/policy-upgrade.feature

```diff
@@ -477,3 +477,181 @@ Feature: SDN compoment upgrade testing
+    Then the output should not contain "200 OK"
+    Then the step should fail
+    Given I use the "<%= cb.proj0 %>" project
+    When I execute on the "<%= cb.rsyslog_server_pod %>" pod:
+      | grep | -w | verdict=drop | /var/log/messages |
+    Then the output should contain "verdict=drop"
+    """
+
+  # @author asood@redhat.com
+  # @case_id OCP-44978
+  @admin
+  @upgrade-check
+  @4.10 @4.9
+  @aws-ipi @gcp-ipi @vsphere-ipi @azure-ipi @baremetal-ipi @openstack-ipi
+  @aws-upi @gcp-upi @openstack-upi
+  @destructive
+  @network-ovnkubernetes
+  Scenario: Check network policy ACL logging works post upgrade 
+    Given I switch to cluster admin pseudo user
+    Given the env is using "OVNKubernetes" networkType
+    And evaluation of `"rsyslog-server"` is stored in the :proj0 clipboard
+    And evaluation of `"policy-upgrade6"` is stored in the :proj1 clipboard
+    And evaluation of `"policy-upgrade7"` is stored in the :proj2 clipboard
+    Given I use the "<%= cb.proj0 %>" project
+    Given I wait for the "rsyslogserver" service to become ready
+    Given a pod becomes ready with labels:
+      | appname=rsyslogserver |
+    And evaluation of `service("rsyslogserver").ip` is stored in the :svc_ip clipboard
+    And evaluation of `pod(0).name` is stored in the :rsyslog_server_pod clipboard
+    Then the step should succeed
+
+    Given as admin I successfully merge patch resource "networks.operator.openshift.io/cluster" with:
+      | {"spec":{"defaultNetwork":{"ovnKubernetesConfig":{"policyAuditConfig": {"destination": "udp:<%= cb.svc_ip %>:514" }}}}} |
+    And I wait up to 300 seconds for the steps to pass:
+    """
+      Given the status of condition "Progressing" for network operator is :False
+      And the status of condition "Available" for network operator is :True
+    """
+
+    Given I use the "<%= cb.proj1 %>" project
+    Given a pod becomes ready with labels:
+      | name=test-pods |
+    And evaluation of `pod(1).ip_url` is stored in the :p1pod1ip clipboard
+    Given I use the "<%= cb.proj2 %>" project
+    Given a pod becomes ready with labels:
+      | name=test-pods |
+    And evaluation of `pod(2).name` is stored in the :p2pod1name clipboard
+    Given a pod becomes ready with labels:
+      | name=hello-idle |
+    And evaluation of `pod(3).name` is stored in the :p2pod2name clipboard
+    And I wait up to 20 seconds for the steps to pass:
+    """
+    Given I use the "<%= cb.proj2 %>" project
+    When I execute on the "<%= cb.p2pod1name %>" pod:
+      | curl | -I | <%= cb.p1pod1ip %>:8080 | --connect-timeout | 5 | 
+    Then the output should contain "200 OK"
+    Then the step should succeed
+    Given I use the "<%= cb.proj0 %>" project
+    When I execute on the "<%= cb.rsyslog_server_pod %>" pod:
+      | grep | -w | verdict=allow | /var/log/messages |
+    Then the output should contain "verdict=allow"
+    """
+    
+    And I wait up to 20 seconds for the steps to pass:
```

Comment:
can we make line 633 and 646 in one loop `And I wait up to 20 seconds for the steps to pass:`

Reply:
@zhaozhanqi That is how I started with one block but Anurag insisted on having two blocks, I agreed as two steps in each block are related i.e with curl there ACL messages would not logged.

Reply:
hmm,  if using two blocks, I think we better remove this part from the block
```
    Given I use the "<%= cb.proj2 %>" project
    When I execute on the "<%= cb.p2pod2name %>" pod:
      | curl | -I | <%= cb.p1pod1ip %>:8080 | --connect-timeout | 2 |
    Then the output should not contain "200 OK"
    Then the step should fail
```
the block only contain
```
And I wait up to 20 seconds for the steps to pass:
 """
    When I execute on the "<%= cb.rsyslog_server_pod %>" pod:
      | grep | -w | verdict=drop | /var/log/messages |
    Then the output should contain "verdict=drop"
    """
```

Reply:
@zhaozhanqi curl request causes these ACL messages to be logged so we need the request.

Reply:
we should not make `    Given I use the "<%= cb.proj2 %>" project` in the loop. 

the is my suggestion:
```
    Given I use the "<%= cb.proj2 %>" project
    When I execute on the "<%= cb.p2pod1name %>" pod:
      | curl | -I | <%= cb.p1pod1ip %>:8080 | --connect-timeout | 5 | 
    Then the step should succeed
    Then the output should contain "200 OK"
    Given I use the "<%= cb.proj0 %>" project
    And I wait up to 20 seconds for the steps to pass:
    """
    When I execute on the "<%= cb.rsyslog_server_pod %>" pod:
      | grep | -w | verdict=allow | /var/log/messages |
    Then the step should succeed
    Then the output should contain "verdict=allow"
    """

    Given I use the "<%= cb.proj2 %>" project
    When I execute on the "<%= cb.p2pod2name %>" pod:
      | curl | -I | <%= cb.p1pod1ip %>:8080 | --connect-timeout | 2 |
    Then the output should not contain "200 OK"
    Then the step should fail

    Given I use the "<%= cb.proj0 %>" project
    And I wait up to 20 seconds for the steps to pass:
    """
    When I execute on the "<%= cb.rsyslog_server_pod %>" pod:
      | grep | -w | verdict=drop | /var/log/messages |
    Then the step should succeed
    Then the output should contain "verdict=drop"
    """
```

Reply:
Need to have curl in the block so that it can be retried if it fails. if it fails there is no point waiting for syslog messages as they will not be any.

I am not sure how switching the project can be avoided in this case.

Reply:
I do not think we need to add curl into loop to retry since we have `--connect-timeout`,  we can set to 20 if 5 is not enough.

Reply:
-connect-timeout does not ensure retry it is only one request with retry success of the test is ensured.

Reply:
I think with --connect-timeout is better and enough. 
we need to make sure one session can be logged.   if curl many times.  how to make sure the first session is logged.

Reply:
This is same suggestion as https://github.com/openshift/verification-tests/pull/2507#discussion_r759538832. @asood-rh Why dont you try with same logic and report any issues here. Then we can go from there i guess

Reply:
@anuragthehatter There is suggestion about having all the steps in one block as well from which it is being insisted to move away. The concern is about switching the project in the loop. As stated earlier curl needs to be retried to ensure ACL messages are generated. There is consideration to make test reliable not just pass in local run.